editor/metaboxes/panel.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { Button, Dashicon } from '@wordpress/components';
+
+class MetaboxPanel extends Component {
+	constructor( props ) {
+		super( ...arguments );
+		this.state = {
+			opened: props.initialOpen === undefined ? true : props.initialOpen,
+		};
+		this.toggle = this.toggle.bind( this );
+	}
+
+	toggle( event ) {
+		event.preventDefault();
+		if ( this.props.opened === undefined ) {
+			this.setState( ( state ) => ( {
+				opened: ! state.opened,
+			} ) );
+		}
+
+		if ( this.props.onToggle ) {
+			this.props.onToggle();
+		}
+	}
+
+	render() {
+		const { title, children, opened } = this.props;
+		const isOpened = opened === undefined ? this.state.opened : opened;
+		const icon = `arrow-${ isOpened ? 'down' : 'right' }`;
+		const className = classnames( 'components-panel__body', { 'is-opened': isOpened } );
+
+		return (
+			<div className={ className }>
```

Comment:
We would need a way for panel to always render its children with a given option so and make them hidden instead of not rendering them. Otherwise with the iframe, we will keep making requests over and over again.

Reply:
> Feels like this could become hard to maintain...

I can understand that, but at the same time this is somewhat just specific to the meta box iframes. Adding extra conditional logic to the PanelBody would pose a different type of difficulty to maintain. We will always have to consider the edge case in which iframes need to be handled, if we were to make them one component.